yt/utilities/parallel_tools/parallel_analysis_interface.py

```diff
@@ -326,6 +326,32 @@ def barrierize(*args, **kwargs):
     return barrierize
 
 
+def parallel_root_only_then_broadcast(func):
```

Comment:
one note: I think we could instead make `parallel_root_only` accept a `broadcast_result` boolean so you could use it like:

```python
@parallel_root_only(broadcast_result=True)
def my_root_only_func():
     .... 
```
and have the default behavior continue to broadcast the `all_clear` integer that it does. Might be better than adding this new function?